### PR TITLE
week 1 questions answered

### PR DESCRIPTION
**Pull request by alfosco**
@s-espinosa
@Carmer

### CODE REVIEW

File: module-2/backend/week_one.md [hunk 1]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
+
+        Post - submits data to a specified resource
+        Get - retrieves data from a specified resource
+        Put - updates a resource completely through a specified resource
+        Delete - deletes a specified resource
+        I believe the fifth verb is 'Patch' and it does the same thing as 'Put'.
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: 🎉 Some people will tell you that PATCH is supposed to be used when you're changing only *part* of a resource (vs. PUT to replace the whole thing), but that generally hasn't come into play.

File: module-2/backend/week_one.md [hunk 2]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+        Delete - deletes a specified resource
+        I believe the fifth verb is 'Patch' and it does the same thing as 'Put'.
+        
 2. What is Sinatra?
+
+        A DSL that enables you to easily create web applications in Ruby.
+        
 4. What is MVC?
+
+        It's an acronym for Model, View, Controller. This is a software design pattern for implementing user interfaces on computers.
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: 👍

File: module-2/backend/week_one.md [hunk 3]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+        
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+
+        It simplfifies desicion making when creating an application. There is no need to waste time on how you will name your paths because that convention is already implemented in Sinatra. You can of course deviate from this convention but it will require you to write more code.
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: 👍

File: module-2/backend/week_one.md [hunk 4]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
 6. What types of variables are accessible in our view templates without explicitly passing them?
+
+        All of the corresponding model's attributes and methods.
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: This is true, but how do you get the model into the view? Generally, in order to maintain a separation of responsibilities we *collect* our data in the controller. Then you can use an **instance variable** to pass it to the view (as you're doing in question 7 below).

File: module-2/backend/week_one.md [hunk 5]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+  @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+
+  ```ruby
+  get '/horses/:name' do
+    erb :index, :locals => {:name => params[:name]}
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: 🎉

File: module-2/backend/week_one.md [hunk 6]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
+13. What does HTTP stand for?
+
+        Hypertext Transfer Protocal.
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+        <% %> Runs the Ruby inside of it but doesn't display it
+        <%= %> Displays the Ruby that is being run inside of it
+
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: 👍

File: module-2/backend/week_one.md [hunk 7]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+        Updates a specific restaurant.
+
+get '/restaurants/:id'
+
+        Displays a specific restaurant.
+
+delete '/restaurants/:id'
+
+        Deletes a specific restaurant.
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: Great!

File: module-2/backend/week_one.md [hunk 8]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
+18. What's a migration?
+
+        A migration alters your database schema over time. Each migration modifies your schema to add or remove tables, columns, or entries.
+
 19. When you create a migration, does it automatically modify your database?
+
+        I believe it only modifies your database if you format your migration table for new changes.
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: True to some degree. The more important point for us at this time is that migrations don't change your database unless you run `rake db:migrate`.

File: module-2/backend/week_one.md [hunk 9]
```diff
@@ -5,30 +5,125 @@ Fork this respository. Answer the questions to the best of your ability. Try to
...
+
 20. How does a model relate to a database?
+
+        A model is mapped to its corresponding table in the database. This enables you to have the ability to map the columns of each row in that table with the attributes of the instances of your model.
```
Comment (s-espinosa) [inline module-2/backend/week_one.md]: 👍

**Comment by alfosco** [comment]
I appreciate the feedback!

**Pull request closed without merging**